app/controllers/events_controller.rb

```diff
@@ -30,16 +30,34 @@ def create
     event = create_event(find_bridge)
     if event.save
       EventWorker.perform_async(event.id)
-      render json: {}, status: 202 # Accepted
+      render json: { id: event.id }, status: 202 # Accepted
     else
       render json: { error: 'Invalid parameters' }, status: 400 # Bad Request
     end
   rescue JSON::ParserError
     render json: { error: 'Invalid request. Payload must be in JSON' }, status: 400 # Bad Request
   end
 
+  def abort
+    events = if bridge_id_present
+               Event.includes(:bridge).where(bridge_id: event_params[:bridge_id], "bridges.user_id": @current_user.id)
```

Comment:
You need to include in the where statement "completed: false". Right now, this looks like it would find all events regardless if they are completed or not.